### DISCUSSION TITLE
Outline won't start after upgrade

### DISCUSSION THREAD

**Discussion by ignitedPotato** [outline/outline#4935] [Self Hosting] [answered]
Outline refuses to start with following error message:
```
outline-dockerized-outline-1  | {"level":"warn","message":"\nThis version of Outline cannot start until a data migration is complete.\nBackup your database, run the database migrations and the following script:\n\n$ node ./build/server/scripts/20210226232041-migrate-authentication.js\n"}
outline-dockerized-outline-1  | error Command failed with exit code 1.
```

I have run the migrations. The message still pops up. The script in question doesn't exist.

**Outline:**
- Install: self hosted
- Version: image ID: 8f65321c7079

**Comment by tvarsis** [comment]
@ignitedPotato  I also saw similar issue with these error logs:
```
error Command "sequelize:migrate" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
yarn run v1.22.18
```

It was due to a breaking change in 0.67.x and mentioned here in release notes:

`Note: sequelize:migrate is no longer included as a command, if you were using this then use db:migrate instead.`

When I changed that, it started working for me.

Maybe you also have some references to old migration type?

**Comment by ignitedPotato** [comment]
@tvarsis I used the following command to migrate:
`docker-compose run --rm outline yarn db:migrate --env=production-ssl-disabled` as stated in the docs.

It still wants me to run `20210226232041-migrate-authentication.js` which doesn't exist.

**Reply by tommoor** [reply]
The migration doesn't exist anymore – that migration was from Feb 2021. You should first update to an older version that includes the migration and then to the latest in 2 steps.

**Reply by ignitedPotato** [reply]
I don't fully understand why that's necessary. My installation is just about 4 weeks old.. 
I'd expect that migration to be present if it's needed for current installations..

**Reply by ignitedPotato** [reply]
Also, I've tried running the script from an older version. This just results in more errors:

```
(node:1) UnhandledPromiseRejectionWarning: SequelizeDatabaseError: column user.service does not exist
    at Query.formatError (/opt/outline/node_modules/sequelize/lib/dialects/postgres/query.js:386:16)
    at Query.run (/opt/outline/node_modules/sequelize/lib/dialects/postgres/query.js:87:18)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async /opt/outline/node_modules/sequelize/lib/sequelize.js:619:16
    at async PostgresQueryInterface.select (/opt/outline/node_modules/sequelize/lib/dialects/abstract/query-interface.js:938:12)
    at async Function.findAll (/opt/outline/node_modules/sequelize/lib/model.js:1753:21)
    at async work (/opt/outline/build/server/scripts/20210226232041-migrate-authentication.js:26:19)
    at async main (/opt/outline/build/server/scripts/20210226232041-migrate-authentication.js:87:3)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:1) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by ignitedPotato** [comment]
The issue seems to be with the following line:
https://github.com/outline/outline/blob/e754f89e5c96def8b830f63b29b2fa779c7ded1f/server/utils/startup.ts#L57

Couldn't this check be simply removed?

I don't have an SSO provider, one team and users who only login via magic link.
This should be possible and not actively prevented. Seeing as the proposed solution also leads to nothing.

**Comment by ignitedPotato** [comment, marked as answer]
I solved the issue by adding a dummy authentication provider in the database for my team.

**Reply by acataluddi** [reply]
Many thanks Maximilian.

To elaborate further your solution for anyone facing the same issue:

1. Connect to PostgreSQL and retrieve your Team ID by the following query:
   ```sql
   select id, name from teams;
   ```

2. Copy your Team ID and replace `__your_team_id_here__` in the query below with it. Run the query.
   ```sql
   insert into authentication_providers
   ("id", "name", "providerId", enabled, "teamId", "createdAt")
   values 
   ('572a6a27-8e58-43c2-8c64-09f64a6e4a5a', 'Dummy', 'bda5cdfa-ac9c-401b-9f0e-c1d8b1a07d7a', true,
     '__your_team_id_here__', now());
   ```

3. You Outline instance should start without problems.

Thanks all for this great product.
Adriano C.
